modules/openapi-generator/src/main/resources/powershell/api_client.mustache

```diff
@@ -91,13 +91,15 @@ function Invoke-{{{apiNamePrefix}}}ApiClient {
 
     {{#hasHttpSignatureMethods}}
     # http signature authentication
-    if ($null -ne $Configuration['ApiKey'] -and $Configuration['ApiKey'].Count -gt 0) {
+    $httpSigningConfig = Get-{{apiNamePrefix}}ConfigurationHttpSigning
+    if ($null -ne $httpSigningConfig) {
         $httpSignHeaderArgument = @{
             Method     = $Method
             UriBuilder = $UriBuilder
             Body       = $Body
+            RequestHeader = $HeaderParameters
         }
-        $signedHeader = Get-{{{apiNamePrefix}}}HttpSignedHeader @httpSignHeaderArgument
+        $signedHeader = Get-{{apiNamePrefix}}HttpSignedHeader @httpSignHeaderArgument
```

Comment:
FYI. {{{ ... }}} will use the original value while {{...}} will use the HTML-escaped value.